[IS-23-24] paso nivel 2 tasapp

AndreaMGG opened the pull request:
Creación fichero is.yaml + creación issue elección del lenguaje de programación

* [x] ¿Todos los mensajes de commit explican el cambio, y no se
      limitan a repetir el nombre del fichero que se ha cambiado?
* [x] ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
* [x] ¿Se ha hecho una revisión real del código para comprobar que todos
   los atributos y funciones creadas están respaldadas por una HU?
* [x] ¿Todos los issues creados están relacionados con a una HU?
* [x] ¿Ha asignado el propietario/la propietaria del repositorio todos los
issues planteados al *milestone* correspondiente?
* [x] ¿Todos los cambios en el código están asignados a un issue al que se
      referencia en un commit?
* [x] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
* [x] ¿Es el milestone sobre el que estoy trabajando un producto
   mínimamente viable? ¿O tengo que solicitar al product manager que precise de
   qué producto se trata?
* [x] ¿Se ha comprobado que el código entregado sea sintácticamente
correcto?


closes #32,
closes #35,
closes #36,
closes #37,
closes #38,
closes #39,
closes #45,
closes #46,
closes #47,
closes #48,
closes #49,
closes #50,
closes #51,
closes #52,
closes #53,
closes #55,
closes #57,
closes #59,
closes #61,
closes #62,
closes #64,
closes #65

sergiohut committed:
Create is.yaml

AndreaMGG commented:
@JJ listo para revisión definitiva

AndreaMGG committed:
is.yaml

sergiohut committed:
Create entidad.py

PabloES796 committed:
recetas.py

sergiohut committed:
Create recetas_tradicionales.py

sergiohut committed:
Delete models/entidad.py

tomasjpardo committed:
Create ingredientes.py

tomasjpardo committed:
Update ingredientes.py

sergiohut committed:
Update recetas_tradicionales.py

sergiohut committed:
Update recetas_tradicionales.py

AndreaMGG committed:
Create preparacion.py

Avz20 committed:
Create ingredientes.py

AndreaMGG committed:
Delete models/entidad.py

Avz20 committed:
Update ingredientes.py

AndreaMGG committed:
Delete models/preparacion.py

tomasjpardo committed:
Update and rename ingredientes.py to unidades.py

PabloES796 committed:
Delete models/entidad.py

AndreaMGG committed:
Renombrar carpeta a 'nueva_carpeta'

sergiohut committed:
renombrar carpeta models

tomasjpardo committed:
Renombrar models a carpeta Recipeers

PabloES796 committed:
cambio nombre carpeta models por recipeers

AndreaMGG committed:
kjdjhc

AndreaMGG committed:
edicion codigo issue5

tomasjpardo committed:
Update unidades.py

se ha creado una biblioteca con las posibles medidas para usar en una receta de cocina.

AndreaMGG committed:
edicion codigo issue5

sergiohut committed:
creación modelo receta que incluya el atributo booleano tradicional

tomasjpardo committed:
Update unidades.py

sergiohut committed:
modificacion nombre atributos clase receta, issue3

Avz20 committed:
Update ingredientes.py

Avz20 committed:
cambiar nombre

Avz20 committed:
archivos actualizados

tomasjpardo committed:
#43 #45. Crea biblioteca con posibles medidas de recetas

sergiohut committed:
Correciones en el modelo receta [ISSUE-1]

tomasjpardo committed:
#43 #45. Crea biblioteca con medidas para recetas

tomasjpardo committed:
Merge branch 'issue-4' of https://github.com/T-ASAPP/T-asapp into issue-4

tomasjpardo committed:
#43 #45. Crea biblioteca de medidas para recetas

tomasjpardo committed:
#35 Correcciones PR #42 modelo Ingredientes

AndreaMGG committed:
#35 Update codigo Ingredientes =! cantidad

JJ commented:
¿Qué os he dicho sobre trabajar en grupo?

JJ commented:
> Creación fichero is.yaml + creación issue elección del lenguaje de programación
> 
> * [x]  ¿Todos los mensajes de commit explican el cambio, y no se
>   limitan a repetir el nombre del fichero que se ha cambiado?
> * [x]  ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
> * [x]  ¿Se ha hecho una revisión real del código para comprobar que todos
>   los atributos y funciones creadas están respaldadas por una HU?
> * [x]  ¿Todos los issues creados están relacionados con a una HU?
> * [x]  ¿Ha asignado el propietario/la propietaria del repositorio todos los
>   issues planteados al _milestone_ correspondiente?
> * [x]  ¿Todos los cambios en el código están asignados a un issue al que se
>   referencia en un commit?
> * [x]  ¿Se ha asignado al mismo milestone el PR que se ha hecho?
> * [x]  ¿Es el milestone sobre el que estoy trabajando un producto
>   mínimamente viable? ¿O tengo que solicitar al product manager que precise de
>   qué producto se trata?
> * [x]  ¿Se ha comprobado que el código entregado sea sintácticamente
>   correcto?

Por favor, no marquéis las cosas hasta que lo hayáis hecho efectivamente. Esta lista de comprobación debía de serviros como guía para que autónomamente veáis si lo habéis hecho bien y también qué os falta.

sergiohut committed:
#37 listaIngredientes pasa a ser una tupla para incluir cantidades

AndreaMGG committed:
#35 UpdatCodigo Ingredientes cambio 'cantidad' a #44

sergiohut committed:
#37 incluimos definición de las clases involucradas en clase receta

Avz20 committed:
#42 cambio título ingredientes a ingrediente

Avz20 committed:
#42 restituyo ingrediente y :coffin:

sergiohut committed:
#37 Incluimos las importaciones de las clases necesarias, incluidas List y Tuple

sergiohut committed:
Merge pull request #42 from T-ASAPP/issue-2

[ISS2] Ingredientes

AndreaMGG committed:
Merge branch 'nivel2' into issue-5

sergiohut committed:
Merge branch 'nivel2' into issue-3

tomasjpardo committed:
#43 #45. Cambio biblioteca por el set con las unidades mas ususales

AndreaMGG committed:
#36 UpdateCodigo + import codigo #35

JJ commented on recipeers/ingrediente.py:
```diff
@@ -0,0 +1,4 @@
+class Ingrediente:
+  def __init__(self, nombre: set[str], medida: str):
```
¿Por qué el nombre es un conjunto?

JJ replied:
Quizás una forma de incorporar a @maguilerafr sea que evalúe el estado actual de este PR y cree otro PR sobre esta rama corrigiendo algunas cosas, como esta que acabo de indicar.

JJ reviewed (changes requested):
Por otro lado, si estáis usando "type hints" en los constructores, ¿no deberíais indicarlo en vuestras revisiones en el resto de los PRs?

tomasjpardo committed:
#43 #45 Cambio a minúsculas de carpeta y se agregan más medidas para receta

tomasjpardo committed:
#43 #45 Se corrigen comas en el set

tomasjpardo committed:
#43 #45. Unificación medidas a plural

sergiohut committed:
#39 Actualización del modelo tradicional, creamos una clase TipoReceta

tomasjpardo committed:
#43 #45. Se elimina directorio Recipeers

tomasjpardo committed:
#43 #45. Se agrega directorio recipeers en minúscula

sergiohut committed:
Merge pull request #43 from T-ASAPP/issue-4

[ISS4] Cantidades y unidades de medida

JJ reviewed (changes requested):
Esta sería mi revisión si me hubiérais dejado tiempo. Como os he pedido que revertáis, ya puestos lo creáis como issue.
Recordad que lo más importante es que plantéis correctamente los issues como problemas. Estas unidades, ¿cómo se van a usar? ¿Las vas a identificar en el texto de una receta? ¿En ese caso, van a aparecer siempre en plural?

Avz20 committed:
import hecho y nombre cambiado de unidades a medida

sergiohut committed:
Merge pull request #41 from T-ASAPP/issue-5

[ISS5] Preparación

sergiohut committed:
elimino el archivo entidad.py puesto que es un archivo vacio y sin funcionalidad

PabloES796 committed:
Update recetas.py

AndreaMGG committed:
Merge pull request #40 from T-ASAPP/issue-3

[ISS3] Recetas tradicionales

AndreaMGG committed:
Merge pull request #44 from T-ASAPP/issue-1

[ISS1] Recetas

JJ commented on recipeers/medida.py:
```diff
@@ -0,0 +1,13 @@
+    "l",   # Litros
+    "ml",  # Mililitros
+    "unidades",  # Conteo de elementos individuales
+    "cucharadas",  # Cucharadas
```
... o eliminando comentarios posiblemente innecesarios.

JJ commented on recipeers/recetas.py:
```diff
@@ -0,0 +1,11 @@
+                 listaIngredientes: List[Tuple[Ingrediente, float]], 
+                 pasosPreparacion: List[Preparacion]):
+        self.nombre = nombre
+        self.listaIngredientes = listaIngredientes
```
O revisando las convenciones. Aquí estáis usando CamelCase sin embargo en el fichero de más arriba se usa snake_case.

JJ reviewed (changes requested):
Sugerencias para empezar a trabajar. Preferiblemente tratables por separado.

JJ commented:
Hay un serie de comentarios en este PR. Deberíais convertir algunos de estos comentarios (o todos) en issues y empezar a trabajar con ellos.

JJ commented on is.yaml:
```diff
@@ -0,0 +1,2 @@
+lenguaje: node.js
+entidad: src/Recordatorio.js
```
Esta entidad ya no existe.

maguilerafr committed:
Issue #42. Relacionar las diferentes clases con las que estamos trabajando con la entidad principal Receta, para que cada una pueda enviar la información pertinente a la clase Receta y que esta en un futuro pueda devolver a partir del contenido del resto la Receta principal.

PabloES796 committed:
Update is.yaml

se define el unico lenguaje y la entidad recetas.py

Avz20 committed:
#51 set eliminado

maguilerafr committed:
Modificación de recetas

Issue #52. La receta es un objeto valor que se forma con la relación de preparacion.py e ingrediente.py. Se han eliminado las funciones porque en este punto del proyecto no es necesario establecer las mismas, solamente las instancias entre estas

PabloES796 reviewed (commented):
Se ha actualizado el fichero

AndreaMGG committed:
Create recetario.py

maguilerafr committed:
Inicialización de Directorio 

Issue #57. El archivo __init__.py facilita la inicialización y organización de directorios en Python, permitiendo la importación de módulos y la ejecución de código de inicialización.

Avz20 committed:
#51 La variable medida solo puede almacenar unidades de medida válidas del tipo enumerado UnidadMedida.

tomasjpardo committed:
#43 #45. Se modifica código con clase enum en nivel 2

PabloES796 reviewed (approved):
Se modifica el archivo is.yaml con la configuración básica del programa.

JJ commented:
> Se modifica el archivo is.yaml con la configuración básica del programa.

Pablo, no puedes aprobar esto todavía. No sé lo que estabas intentando hacer, pero este PR no se aprueba hasta que esté completo (o se revisa)

AndreaMGG committed:
Borro recetario.py x no seguir buenas practicas commit

Elimino el archivo recetario.py porque el mensaje de commit no seguía las buenas prácticas.

PabloES796 committed:
:coffin:

AndreaMGG committed:
Creo código de entidad recetario del issue #55

Necesitamos el código para la entidad Recetario, así este define una clase recetario que permite almacenar y gestionar recetas. Esta importa el código del archivo recetas_tradicionales para filtrar recetas por tipo y también importa el objeto valor receta, el cual importa a su vez el código de los objetos valor ingredientes, medidas y preparación, todos los componentes de la receta.

AndreaMGG committed:
Merge pull request #58 from T-ASAPP/issue-7

[ISS7] Entidad Recetario

JJ commented on recipeers/recetas.py:
```diff
@@ -0,0 +1,11 @@
+from preparacion import Preparacion
+from ingrediente import Ingrediente
+
+class Receta:
```
¿El nombre de este fichero se corresponde con la clase que contiene?

JJ commented on recipeers/medida.py:
```diff
@@ -0,0 +1,45 @@
+    l = "Litros"
+    ml = "Mililitros"
+    unidades = "Conteo de elementos individuales"
+    cucharadas = "Cucharadas"
```
¿Seguimos usando el plural?

maguilerafr committed:
Merge pull request #54 from T-ASAPP/issue-6

[ISS6] Objeto Valor Receta

JJ commented on recipeers/medida.py:
```diff
@@ -0,0 +1,45 @@
+    vasos_de_yogurt = "Vasos de yogurt (125 ml aprox)"
+    al_gusto = "Al gusto (saonar)"
+
+def agregar_medidas(conjunto_medidas, nuevas_medidas):
```
¿Cómo es que habéis aceptado este código que incluye funciones?

JJ commented on recipeers/medida.py:
```diff
@@ -0,0 +1,45 @@
+    """
+    Agrega nuevas medidas al conjunto de medidas existente.
+
+    :param conjunto_medidas: Conjunto (set) de medidas existente.
```
El conjunto de medidas existente son un enum. Si vais a hacer que este conjunto se pueda expandir, ¿seguro que usar un enum es lo más razonable?

JJ commented on recipeers/medida.py:
```diff
@@ -0,0 +1,45 @@
+    :param nuevas_medidas: Lista de nuevas medidas a agregar.
+    """
+    for nueva_medida in nuevas_medidas:
+        conjunto_medidas.add(nueva_medida)
```
¿Vais a trabajar con una variable global para el conjunto de medidas? ¿Una además que declaráis debajo?

JJ commented on recipeers/medida.py:
```diff
@@ -0,0 +1,45 @@
+nuevas_medidas = []
+
+# Intentar agregar nuevas medidas al conjunto (no habrá cambios)
+agregar_medidas(conjunto_medidas, nuevas_medidas)
```
O es una variable global o es un argumento. No puede ser las dos cosas.

JJ reviewed (changes requested):
No he visto este PR. Quizás podríais reconsiderarlo.

PabloES796 committed:
:coffo

Avz20 committed:
#62 camino correcto

AndreaMGG committed:
update cod recetas,rtradicionales,ingredi

He cambiado el codigo de los archivos recetas.py,recetas__tradicionales.py e ingrediente.py en cuanto a nombre de las clases y variables para que sigan la convecion snakecase ya que trabajamos en python

JJ commented on recipeers/recetas_tradicionales.py:
```diff
@@ -1,6 +1,6 @@
-tradicional = TipoReceta("Tradicional")
\ No newline at end of file
+# Crear instancias de Tipo_Receta
+tradicional = Tipo_Receta("Tradicional")
```
Pero si este estaba bien...

JJ reviewed (commented):
CamelCase no usa el símbolo `_`

PabloES796 committed:
Rename license to LICENSE

PabloES796 committed:
Rename readme.md to README.md

PabloES796 committed:
Merge pull request #60 from T-ASAPP/borrabot

Se borra el archivo Bot

tomasjpardo committed:
#64. Se realizan los cambios solicitados: eliminación del plural, eliminación de funciones, eliminación del uso de enum, eliminación del uso de variable global

AndreaMGG committed:
cambio nombre recetas.py a receta.py

AndreaMGG committed:
cambio recetas_tradicionales.py a tipo_receta.py

AndreaMGG committed:
update cod ingrediente,recetario,tipo_receta.py

He cambiado el codigo de los archivos ingrediente.py, recetario.py y tipo_receta.py en cuanto a los nombres de las clases y las variables y la importacion de los nombres de los ficheros cuyo nombre se ha actualizaado tambien

AndreaMGG committed:
update codigo tipo_receta.py

para corregir un error en el codigo de este fichero en cuanto al nombre de la clase para que siga la convencion snakecase en la que estamos trabajando en python

AndreaMGG committed:
borro recetas_tradicionales.py por duplicidad archivo

PabloES796 committed:
Rename license to LICENSE

PabloES796 committed:
Rename readme.md to README.md

AndreaMGG committed:
actualizo codigo nombre clase TipoReceta tipo_receta.py

Avz20 committed:
#62 eliminar clave configuracion

maguilerafr committed:
Adición de los Docstring de tipo receta. Issue #65

PabloES796 committed:
Update medida.py

Se cambia yogurt por yogur, ya que a la hora de hacer una búsqueda simplifica y tambien sirve para ambos.

maguilerafr committed:
Adición de los Docstring en recetario. Issue #65

maguilerafr committed:
Adición de los Docstring en receta. Issue #65

maguilerafr committed:
Adición de los Docstring en preparacion. Issue #65

maguilerafr committed:
Adición de los Docstring en ingrediente. Issue #65

maguilerafr committed:
Merge pull request #67 from T-ASAPP/nivel2-medida

#64. Se realizan los cambios solicitados en el archivo media.py

maguilerafr committed:
Adición de los Docstring en medidas. Issue #65

Avz20 committed:
Merge pull request #63 from T-ASAPP/issue62

#62 camino correcto

JJ commented on is.yaml:
```diff
@@ -1,3 +1,2 @@
-configuracion:
-  lenguaje: "Español"
-  entidad: recetas.py
+lenguaje: "Español"
```
"Español" es un idioma, no un lenguaje.

JJ reviewed (commented):
¿Veis como hay que revisarlo todo?

maguilerafr committed:
Merge branch 'nivel2' of https://github.com/T-ASAPP/T-asapp into nivel2

maguilerafr committed:
correción docstring ingrediente.py

maguilerafr committed:
correción docstring recetario.py

AndreaMGG committed:
Correcion docstring preparacion.py #65

Avz20 committed:
#65 adición de los docstring

maguilerafr committed:
correción docstring tipo_receta.py Issue #65

AndreaMGG committed:
Merge branch 'nivel2' of https://github.com/T-ASAPP/T-asapp into nivel2

PabloES796 committed:
Update receta.py

Se añade docstring al código.

maguilerafr committed:
Merge branch 'nivel2' of https://github.com/T-ASAPP/T-asapp into nivel2

JJ commented on recipeers/preparacion.py:
```diff
@@ -0,0 +1,32 @@
+
+    """
+
+    def __init__(self, verbos_cocinado: Set[str], ingredientes: str, instrucciones: str):
```
¿Cada instancia de `Preparacion` va a tener un conjunto de verbos diferente?

AndreaMGG replied:
Pensamos que sí

JJ replied:
Vamos a ver. ¿El conjunto de verbos es el conjunto de verbos admisible *en todas las recetas* o cada receta va a tener un conjunto diferente de verbos admisibles?

AndreaMGG replied:
Cada vez que se instancie la clase Preparacion la receta contendrá un conjunto diferente de verbos_cocinado ya que entendemos como instrucciones aquella sentencia que se compone una serie de verbos empleados para cocinar e ingredientes. 
Así pues, cada receta tendrá un conjunto de verbos_cocinado diferente.
De la misma forma, en la clase Ingrediente se conforma un conjunto distinto de ingredientes cada vez que se instancia en cada receta.

JJ reviewed (changes requested):
Por favor, dadle un vistazo a esto.

Avz20 committed:
correción de errata

AndreaMGG commented:
Sobre la estructura del repositorio
* [x] ¿He seguido las mejores prácticas en el nombre de las clases y ficheros y
   disposición de los mismos?
* [x] ¿He mirado tanto reglas generales (para cualquier repo) como reglas
específicas para el lenguaje con el que se está trabajando?
Sobre el planteamiento
* [x] ¿Se han planteado una serie de issues, asignados a las historias de
usuario pertinentes, que clarifiquen qué es lo que necesitan los usuarios e
identifiquen las diferentes partes del problema?
* [x] ¿Los issues representan un problema, y no una tarea?
Sobre el análisis del problema
* [x] ¿Se ha documentado qué análisis se ha hecho sobre el dominio para decir lo
   que se ha creado?
* [x] ¿Se ha documentado por qué se ha elegido que lo creado sea un objeto valor,
   una entidad o un agregado?
* [x] ¿He propuesto un producto mínimamente viable, que en muchos casos será un solo
   objeto valor que no dependa de ningún otro (y que sea la base de muchos
   otros)?.
Sobre el código
* [x] ¿Todos los mensajes de commit explican el cambio, y no se
      limitan a repetir el nombre del fichero que se ha cambiado?
* [x] ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
* [x] ¿Se ha hecho una revisión real del código para comprobar que todos
   los atributos y funciones creadas están respaldadas por una HU?
* [x] ¿Todos los issues creados están relacionados con a una HU?
* [x] ¿Ha asignado el propietario/la propietaria del repositorio todos los
issues planteados al *milestone* correspondiente?
* [x] ¿Todos los cambios en el código están asignados a un issue al que se
      referencia en un commit?
* [x] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
* [x] ¿Es el milestone sobre el que estoy trabajando un producto
   mínimamente viable? ¿O tengo que solicitar al product manager que precise de
   qué producto se trata?
* [x] ¿Se ha comprobado que el código entregado sea sintácticamente
correcto?

JJ commented:
¿Habéis revisado todos los issues del milestone a ver si habéis creado código para resolverlos todos?

maguilerafr commented:
> ¿Habéis revisado todos los issues del milestone a ver si habéis creado código para resolverlos todos?

Hemos revisado y creemos que sí

JJ commented:
¿Y habéis puesto en el cuerpo del PR los issues que se van a cerrar cuando se mergee?

JJ commented:
De la comprobación de sintaxis:

```text
Traceback (most recent call last):
  File "/Users/jjmerelo/.pyenv/versions/3.9.13/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/jjmerelo/.pyenv/versions/3.9.13/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/jjmerelo/Asignaturas/MPDA-IS/T-asapp/recipeers/recetario.py", line 8, in <module>
    from tipo_receta import Tipo_receta
ImportError: cannot import name 'Tipo_receta' from 'tipo_receta' (/Users/jjmerelo/Asignaturas/MPDA-IS/T-asapp/recipeers/tipo_receta.py)
```

maguilerafr committed:
Corrección de errata del nombre de clase

AndreaMGG committed:
Correcion nombre clase TipoReceta recetario.py

AndreaMGG committed:
Merge branch 'nivel2' of https://github.com/T-ASAPP/T-asapp into nivel2

maguilerafr commented:
> De la comprobación de sintaxis:
> 
> ```
> Traceback (most recent call last):
>   File "/Users/jjmerelo/.pyenv/versions/3.9.13/lib/python3.9/runpy.py", line 197, in _run_module_as_main
>     return _run_code(code, main_globals, None,
>   File "/Users/jjmerelo/.pyenv/versions/3.9.13/lib/python3.9/runpy.py", line 87, in _run_code
>     exec(code, run_globals)
>   File "/Users/jjmerelo/Asignaturas/MPDA-IS/T-asapp/recipeers/recetario.py", line 8, in <module>
>     from tipo_receta import Tipo_receta
> ImportError: cannot import name 'Tipo_receta' from 'tipo_receta' (/Users/jjmerelo/Asignaturas/MPDA-IS/T-asapp/recipeers/tipo_receta.py)
> ```

Creíamos que habíamos corregido ese fallo, ya lo hemos hecho definitivamente.

maguilerafr commented:
> ¿Y habéis puesto en el cuerpo del PR los issues que se van a cerrar cuando se mergee?

Ya hemos hecho el listado de los issues a cerrar y lo hemos puesto en la descripción del PR.
Son estos: closes #32, #35, #36, #37, #38, #39, #45, #46, #47, #48, #49, #50, #51, #52, #53, #55, #57, #59, #61, #62, #64, #65

JJ commented:
Adicionalmente
```txt
poetry run pyflakes recipeers/*.py
recipeers/preparacion.py:7:1: 'ingrediente.Ingrediente' imported but unused
recipeers/recetario.py:8:1: 'tipo_receta.Tipo_receta' imported but unused
```

JJ commented:
No sé donde visteis lo de que se podían separar por comas, pero conviene que os acostumbréis a usar la fuente. [En este manual](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) explica como hacerlo, incluso con issues de otro repositorio (si se tiene permiso, claro)

maguilerafr commented:
> No sé donde visteis lo de que se podían separar por comas, pero conviene que os acostumbréis a usar la fuente. [En este manual](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) explica como hacerlo, incluso con issues de otro repositorio (si se tiene permiso, claro)

ok, gracias, le echo un ojo

maguilerafr committed:
Correción clase ingrediente / medida

Dentro de la clase ingrediente, la medida no se estaba usando correctamente, donde a cada ingrediente se vinculaba con toda la lista de medidas y no una medida concreta
